File: KataFive/KataFiveTest.py
import unittest
from KataFive import BloomFilter


class Test(unittest.TestCase):
    spell_checker = BloomFilter('/usr/share/dict/words')

    def test_contains(self):
        bloom_filter = BloomFilter()
        bloom_filter.insert("hello")
        self.assertTrue(bloom_filter.contains("hello"), 'filter should contain "hello"')

    def test_does_not_contains(self):
        bloom_filter = BloomFilter()
        bloom_filter.insert("world")
        self.assertFalse(bloom_filter.contains("hello"), 'filter should not contain "hello"')

    def test_spell_checker_contains(self):
        self.assertTrue(self.spell_checker.contains("hello"), '"hello" should be a word')

    def test_spell_checker_does_not_contain(self):
        word = 'bobloblaw'
        self.assertFalse(self.spell_checker.contains(word), '"' + word + '" should not be a word')


if __name__ == "__main__":
    unittest.main()

File: ruby/05/KataFiveTest.rb
require 'test/unit'
require 'KataFive'

class KataFiveTest < Test::Unit::TestCase
  @@spell_checker = BloomFilter.new('/usr/share/dict/words')

  def test_contains
    word = 'hello'
    bloom_filter = BloomFilter.new()
    bloom_filter.insert(word)
    assert(bloom_filter.contains(word), "filter should contain #{word}")
  end

  def test_does_not_contain
    word = 'hello'
    bloom_filter = BloomFilter.new()
    bloom_filter.insert("#{word} world")
    assert(!bloom_filter.contains(word), "filter should not contain #{word}")
  end

  def test_spelling_contains
    word = 'hello'
    assert(@@spell_checker.contains(word), "#{word} should be a word")
  end

  def test_spelling_does_not_contain
    word = 'bobloblaw'
    assert(!@@spell_checker.contains(word), "#{word} should not be a word")
  end
end
